pina/geometry/simplex.py

```diff
@@ -0,0 +1,243 @@
+import torch
+from .location import Location
+from pina.geometry import CartesianDomain
+from pina import LabelTensor
+from ..utils import check_consistency
+
+class SimplexDomain(Location):
+    """PINA implementation of a Simplex."""
+
+    def __init__(self, simplex_matrix, sample_surface=False):
+        """
+        :param simplex_matrix: A matrix of LabelTensor objects representing
+            a vertex of the simplex (a tensor), and the coordinates of the
+            point (a list of labels).
+        :type simplex_matrix: list[LabelTensor]
+        :param sample_surface: A variable for choosing sample strategies. If
+            `sample_surface=True` only samples on the Simplex surface
+            frontier are taken. If `sample_surface=False`, no such criteria
+            is followed.
+        :type sample_surface: bool
+
+        .. warning::
+            Sampling for dimensions greater or equal to 10 could result
+            in a shrinking of the simplex, which degrades the quality
+            of the samples. For dimensions higher than 10, other algorithms
+            for sampling should be used.
+        
+        :Example:
+            >>> spatial_domain = SimplexDomain(
+                    [
+                        LabelTensor(torch.tensor([[0, 0]]), labels=["x", "y"]),
+                        LabelTensor(torch.tensor([[1, 1]]), labels=["x", "y"]),
+                        LabelTensor(torch.tensor([[0, 2]]), labels=["x", "y"]),
+                    ], sample_surface = True
+                )
+        """
+
+        # check consistency of sample_surface
+        check_consistency(sample_surface, bool)
+        self._sample_surface = sample_surface
+
+        # check consistency of simplex_dict
+        if not isinstance(simplex_matrix, list):
+            raise ValueError(f"{type(simplex_matrix).__name__} must be {list}.")
+        check_consistency(simplex_matrix, LabelTensor)
+
+        # check consistency of labels
+        if simplex_matrix[0].labels != simplex_matrix[1].labels:
+            raise ValueError(f"Labels don't match.")
+        
+        # check consistency of labels
+        if simplex_matrix[0].labels != simplex_matrix[1].labels:
+            raise ValueError(f"Labels don't match.")
+
+        # vertices, vectors, coordinates
+        self._vertices_matrix = torch.tensor(
+            [
+                [float(vertex.extract(label)) for label in vertex.labels]
```

Comment:
why do we need float?

Reply:
`vertex.extract(label)` returns a `LabelTensor` object, so I get a similar error to the one I got in the Slack thread: `ValueError: could not determine the shape of object type 'LabelTensor'`.

Reply:
But why cannot we use simply a LabelTensor matrix. You could call `torch.cat` and then assign the labels... Would this work? I want to avoid to write `float`

Reply:
It should be working now with that fix.